File: 2022/js/days/day04/day04.js
import * as fs from 'fs';
const array = fs.readFileSync('days/day04/input.txt', 'utf-8').trim().split('\n');

let answer1 = 0;
let answer2 = 0;

array.map((value) => {
    let line = value.split(',');
    let int1 = line[0].split('-').map(Number);
    let int2 = line[1].split('-').map(Number);

    //Part 1
    if (int1[0] <= int2[0] && int2[0] <= int1[1] && int1[0] <= int2[1] && int2[1] <= int1[1]) {
        answer1 = answer1 + 1;
    } else if (int2[0] <= int1[0] && int1[0] <= int2[1] && int2[0] <= int1[1] && int1[1] <= int2[1]) {
        answer1 = answer1 + 1;
    }

    //Part 2
    if ((int1[0] <= int2[0] && int2[0] <= int1[1]) || (int1[0] <= int2[1] && int2[1] <= int1[1])) {
        answer2 = answer2 + 1;
    } else if ((int2[0] <= int1[0] && int1[0] <= int2[1]) || (int2[0] <= int1[1] && int1[1] <= int2[1])) {
        answer2 = answer2 + 1;
    }
});

console.log('Part 1 : ' + answer1);
console.log('Part 2 : ' + answer2);

File: 2019/days/day04/day04.py
input = open("days/day04/input.txt", "r").read().split("-")
min = int(input[0])
max = int(input[1])

# Part 1
numbers = [i for i in range(min, max + 1) if list(str(i)) == sorted(str(i)) and len(set(str(i))) < len(str(i))]
print(f"First star solution : {len(numbers)}")

# Part 2
numbers = [i for i in numbers if 2 in [str(i).count(j) for j in set(str(i))]]
print(f"Second star solution: {len(numbers)}")

